src/pykeen/nn/representation.py

```diff
@@ -1649,9 +1753,10 @@ def __init__(
             ).shape[1:],
             reference=shape,
         )
-        # infer max_id
-        max_id = max_id or base.max_id
-        if max_id != base.max_id:
+        if max_id is None:
+            # infer max_id
+            max_id = base.max_id
+        elif max_id != base.max_id:
             raise ValueError(f"Incompatible max_id={max_id} vs. base.max_id={base.max_id}")
```

Comment:
```suggestion
            raise ValueError(f"Incompatible {max_id=:_} vs. {base.max_id=:_}")
```

Maybe it makes sense to introduce a custom exception for those `max_id` mismatch cases?

Reply:
Done in #1508